src/comments.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import json
+import argparse
+import os
+import requests
+import sys
+from xmlschema import XMLSchema
+from pprint import pprint
+
+try:
+    from .providers import find_ci_provider
+except ImportError:
+    from providers import find_ci_provider
+
+
+try:
+    from .utils import EnvDefault, valid_repo_name, valid_token, valid_pr, valid_file, valid_sha
+except ImportError:
+    from utils import EnvDefault, valid_repo_name, valid_token, valid_pr, valid_file, valid_sha
+
+
+GITHUB_BASE = os.environ.get("GITHUB_BASE", "https://api.github.com")
+GITHUB_PR_COMMENTS_URL = GITHUB_BASE + "/repos/{repo_name}/pulls/{pull_request}/comments"
+
+ACTIVE_CI = find_ci_provider()
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__name__)
+
+
+def update_comments(junit = None, repo_name = None, pull_request = None, sha = None, token = None):
+    url = GITHUB_PR_COMMENTS_URL.format(repo_name=repo_name, pull_request=pull_request, token=token)
+    headers = {"Content-Type": "application/json"}
+    """
+    params = dict()
+    params['direction']='asc'
+    params['sort']='created'
+    result = requests.get(url, data=json.dumps(params), headers=headers)
+    body = json.loads(result.content.decode('utf8'))
+    print(len(body))
+    pprint(body)
+    """
+
+    payload  = dict()
+    matcher = r".*:\d{1,}:\d{1,}:\ .*"
```

Comment:
./src/comments.py:44:5: F841 local variable 'matcher' is assigned to but never used